servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -245,34 +250,21 @@ public void commitBlock(long sessionId, long blockId)
    * @throws WorkerOutOfSpaceException if this Store has no more space than the initialBlockSize
    * @throws IOException if blocks in eviction plan fail to be moved or deleted
    */
-  public String createBlock(long sessionId, long blockId, int tierAlias, long initialBytes)
-      throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException {
-    BlockStoreLocation loc =
-        tierAlias == -1 ? BlockStoreLocation.anyTier() : BlockStoreLocation.anyDirInTier(tierAlias);
-    TempBlockMeta createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, loc, initialBytes);
-    return createdBlock.getPath();
-  }
-
-  /**
-   * Creates a block. This method is only called from a data server.
-   *
-   * Call {@link #getTempBlockWriterRemote(long, long)} to get a writer for writing to the block.
-   *
-   * @param sessionId The id of the client
-   * @param blockId The id of the block to be created
-   * @param tierAlias The alias of the tier to place the new block in, -1 for any tier
-   * @param initialBytes The initial amount of bytes to be allocated
-   * @throws IllegalArgumentException if location does not belong to tiered storage
-   * @throws BlockAlreadyExistsException if blockId already exists, either temporary or committed,
-   *         or block in eviction plan already exists
-   * @throws WorkerOutOfSpaceException if this Store has no more space than the initialBlockSize
-   * @throws IOException if blocks in eviction plan fail to be moved or deleted
-   */
-  public void createBlockRemote(long sessionId, long blockId, int tierAlias, long initialBytes)
-      throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException {
-    BlockStoreLocation loc = BlockStoreLocation.anyDirInTier(tierAlias);
-    TempBlockMeta createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, loc, initialBytes);
-    FileUtils.createBlockPath(createdBlock.getPath());
+  public String createBlock(long sessionId, long blockId, long initialBytes)
+      throws BlockAlreadyExistsException, WorkerOutOfSpaceException, BlockDoesNotExistException,
+      IOException, InvalidWorkerStateException {
+    // add flag check here?
+    BlockStoreLocation loc = BlockStoreLocation.anyDirInTier(StorageLevelAlias.MEM.getValue());
+    TempBlockMeta createdBlock;
+    try {
+      createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, loc, initialBytes);
+      return createdBlock.getPath();
+    } catch (WorkerOutOfSpaceException e) {
+      LOG.info("MEM tier is full, try to create block in any tier");
+      createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, BlockStoreLocation.anyTier(),
```

Comment:
hmm, I see. I thought for (2) `createBlockMeta` will move the evicted blocks down to sdd or hdd, and give higher priority to the latest request. It seems there are two related factors: (1) whether evict other blocks in the tier (2) whether try lower-tier storage, and therefore may result in different combinations: (1) evict other blocks first, if full, try lower-tier storage (2) do not evict other blocks, but try lower-tier storage (3) evict other blocks, but do not try lower-tier storage (4) do not evict other blocks, nor try lower-tier storage, but throw exception.

So other than having a built-in policy, we may take different policies that users want, and explain it in details. What do you think?